apps/labs/posts/jupyter-everywhere.md

```diff
@@ -0,0 +1,229 @@
+<!-- link to jupyterlite draft PR for custom export plugin here -->
+
+`jsPDF` is a JavaScript library that allows generating PDF documents directly in the browser, without server-side processing. We integrated `jsPDF` into Jupyter Everywhere by creating a custom download button that converts the current notebook to a PDF and triggers its download. However, this functionality is still a work in progress. There is work to be done, viz., improving the fidelity of the generated PDFs, especially for complex notebooks with rich visualisations, widgets, and typeset LaTeX content.
+
+#### Uploading notebooks to Jupyter Everywhere
+
+Since downloading notebooks is only half the story, you might ask – how does one upload notebooks that might have been created elsewhere (whether in Jupyter Notebook, JupyterLab, or another JupyterLite instance, or Jupyter Everywhere itself), into Jupyter Everywhere?
+
+Remember that Jupyter Everywhere does not have user accounts or a backend database to store user data and notebooks grouped by accounts. The frontend runs in what is termed the "Simple Interface", or more technically, the "single-document mode" of JupyterLab, which we customise heavily by disabling and reimplementing Jupyter extensions to provide a more user-friendly experience. This means there is no visible file browser or file management interface in Jupyter Everywhere, to avoid unnecessary complexity.
+<!-- link to simple interface docs in jupyter -->
+
+Hence, users need to upload notebooks manually to a JupyterLite instance before the Jupyter Everywhere session is instantiated. The main entry point for this is the Jupyter Everywhere landing page, where we've added an "Upload a Notebook" button that lets users select a local IPyNB file from their computer and upload it to the JupyterLite file system.
+
+Here is a quick walkthrough of how it works behind the scenes.
+<!-- add pictures -->
+
+When the user clicks the "Upload a Notebook" button, a file input dialogue is opened, allowing the user to select a local IPyNB file from their computer. Once the file is selected and uploaded, we use a feature in modern browsers, the Web Storage API, to store the contents of the uploaded notebook in `localStorage` without interacting with the sharing service or requiring the user's explicit consent.
+
+We then redirect the user away from the Jupyter Everywhere landing page and to the JupyterLite application URL, appending a query parameter to the UUID of the uploaded notebook. This query parameter is then parsed by Jupyter Everywhere upon startup. The notebook is read from `localStorage` and written to the JupyterLite in-browser file system, allowing the user to open and interact with the uploaded notebook as if it were created in Jupyter Everywhere itself. The notebook is then removed from `localStorage` to free up space, and no race conditions occur because the notebook is read only once during startup.
```

Comment:
These paragraphs provide a very detailed description of a workaround for a problem, but the problem is not stated in the first place.

Reply:
Added in c7671f0d5cb8914139ece7299c7d9a2c903f9f42, PTAL!